libs/ui/src/components/NeoSelect/NeoSelect.vue

```diff
@@ -1,36 +1,78 @@
 <template>
-  <o-field :label="label">
-    <o-select :placeholder="placeholder">
-      <option
-        v-for="(option, index) in options"
-        :key="index"
-        :value="option.value">
-        {{ option.text }}
-      </option>
-    </o-select>
-  </o-field>
+  <o-select
+    v-model="selected"
+    :placeholder="placeholder"
+    class="neo-select"
+    :class="{ 'o-ctrl-sel--expanded': expanded }"
+    icon-right="chevron-down">
+    <option
+      v-for="(option, index) in selectOptions"
+      :key="index"
+      :value="option.value">
+      {{ option.text }}
+    </option>
+  </o-select>
 </template>
 
-<script lang="ts" setup>
-import { OField, OSelect } from '@oruga-ui/oruga'
+<script setup>
```

Comment:
is there any concern if we using mixins instead here? should be easier to migrate `b-select`, and the test should be ok

```vue
<script>
import { OSelect } from '@oruga-ui/oruga'

export default {
  mixins: [OSelect],
  computed: {
    rootClasses() {
        return [
          'neo-select',
            this.computedClass('rootClass', 'o-ctrl-sel'),
            { [this.computedClass('expandedClass', 'o-ctrl-sel--expanded')]: this.expanded },
        ]
    },
  },
}
</script>

<style lang="scss">
@import './NeoSelect.scss';
</style>

```

Reply:
> Clicking at the arrow wont open the dropdown: (General)

I don't have this issue with extending `OSelect` components

Reply:
are `selectClasses` imported too?
I'm ok with that but will https://github.com/kodadot/nft-gallery/pull/6200#issuecomment-1587472284 still work?

Reply:
And second problem is that now we need to provide select options everywhere within HTML... I don't like it but it's up to others how they want it

Reply:
we can use `iconRight` props to change the icon. but unable to click it to open the dropdown by default

https://github.com/kodadot/nft-gallery/assets/734428/c193c4b0-2e1e-4159-b19a-0d0476a3be56

Reply:
Yes we know that already :D :D

Reply:
> Yes we know that already :D :D

if that is the case, mixins were easier, isn't it?
or did I miss any? what is the issue with `selectClasses`?

Reply:
What? 😆 What does the problem with custom icon and mixin have in common?

Reply:
I mean, I prefer to extend Oruga Select in NeoSelect. code snippet here: https://github.com/kodadot/nft-gallery/pull/6200#discussion_r1227695561

with that, no need to fix the failed unit test

Reply:
and, we dont have the other issue mentioned by `prury`

for the icon, we also can change that with css if we dont want custom icon
![image](https://github.com/kodadot/nft-gallery/assets/734428/ae053363-339e-4bd9-b352-332fa59f624c)

Reply:
Okay but still oruga select doesnt have options prop so you would need to manually add options everywhere, but my solution provides prop to set options with provided array and arrow from mixin doesnt work in dark mode

Reply:
> Okay but still oruga select doesnt have options prop so you would need to manually add options everywhere

yes, maybe this is Oruga principle to go that way https://github.com/oruga-ui/oruga/blob/develop/.github/CONTRIBUTING.md#2-lightweight-and-simple for now

> arrow from mixin doesnt work in dark mode

you can add this CSS to fix the icon on dark mode

```css
.dark-mode .neo-select .o-sel-arrow {
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' height='1em' viewBox='0 0 512 512'%3E%3C!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2023 Fonticons, Inc. --%3E%3Cstyle%3Esvg%7Bfill:%23ffffff%7D%3C/style%3E%3Cpath d='M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z'/%3E%3C/svg%3E");
}

.light-mode .neo-select .o-sel-arrow {
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' height='1em' viewBox='0 0 512 512'%3E%3C!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2023 Fonticons, Inc. --%3E%3Cstyle%3Esvg%7Bfill:%23000000%7D%3C/style%3E%3Cpath d='M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z'/%3E%3C/svg%3E");
}
```

Reply:
imo, if the component already exists on Oruga, just use their default behaviour first by extending their component. to make it easier, because our goal is to migrate from buefy to oruga first

if we want to introduce another behaviour or even a better solution. it's good, but we can do it after migration or put it in the other custom component. in this case, we can create separate PR and then create another component `<NeoSelectOption />` with your solution

another example is that we still have `b-navbar` that doesnt have an oruga component. we can copy buefy implementation, or maybe create our own component

Reply:
> imo, if the component already exists on Oruga, just use their default behaviour first by extending their component. to make it easier, because our goal is to migrate from buefy to oruga first

Yes we first introduce breaking changes because we missed important wrapping class (#6197)
Let's first focus on migration and then we can improve some components.

> another example is that we still have `b-navbar` that doesnt have an oruga component. we can copy buefy implementation, or maybe create our own component

https://github.com/oruga-ui/theme-bulma#buefy-users "Soon" ahah but I think custom Bulma navbar is the way.